libs/map/src/lib/components/layer-switcher/vl-map-layer-switcher.ts

```diff
@@ -1,134 +1,144 @@
-        this._addMapListener();
-    }
 
-    _nonBaseLayersReady() {
-        return Promise.all(this._nonBaseLayers.map((layer) => awaitUntil(() => layer.ready)));
+        return Promise.all(this.mapElement?.nonBaseLayers.map((layer) => awaitUntil(() => layer.ready)));
     }
 
-    _addChangeListeners() {
-        this._layerInputs.forEach((input) => {
-            this._initializeInput(input);
-            input.addEventListener('change', () => this._setLayerVisibility(input));
-        });
-    }
-
-    _addMapListener() {
-        this._mapElement.on('moveend', () => this._computeInputsDisabledAttribute());
-    }
-
-    _initializeInput(input) {
-        const layer = this._getLayer(input);
-        if (layer) {
-            input.checked = layer.visible;
+    private getVlMapLayers(): VlMapLayer[] {
+        if (!this.layers || !Array.isArray(this.layers)) {
+            return this.mapElement?.nonBaseLayers || [];
         }
-    }
 
-    _setLayerVisibility(input) {
-        const layer = this._getLayer(input);
-        if (layer) {
-            layer.visible = input.checked;
-            this._mapElement.rerender();
-        }
+        return this.mapElement?.nonBaseLayers.filter((layer) => this.layers?.includes(layer.name)) || [];
     }
 
-    _computeInputsDisabledAttribute() {
-        this._layerInputs.forEach((input) => this._computeInputDisabledAttribute(input, this._mapElement.resolution));
-    }
+    private computeDisabledAttributes = () => {
+        const resolution = this.mapElement?.resolution;
+
+        this.vlMapLayers.forEach((layer) => {
+            const checkbox = this.shadowRoot?.querySelector(`vl-checkbox[data-vl-layer="${layer.title}"]`);
 
-    _computeInputDisabledAttribute(input, resolution) {
-        const layer = this._getLayer(input);
-        if (layer) {
-            if (layer.isVisibleAtResolution(resolution)) {
-                input.removeAttribute('disabled');
+            if (!layer.isVisibleAtResolution(resolution)) {
+                checkbox?.setAttribute('disabled', '');
             } else {
-                input.setAttribute('disabled', '');
+                checkbox?.removeAttribute('disabled');
             }
-        }
+        });
+    };
+
+    private observeMapLayers() {
+        this.layerObserver = new MutationObserver((mutations) => {
+            mutations.forEach((mutation) => {
+                mutation.addedNodes.forEach((node) => {
+                    if ((node as HTMLElement).hasAttribute('data-vl-is-layer')) {
+                        this.vlMapLayers = [...this.vlMapLayers, node as unknown as VlMapLayer];
+                    }
+                });
+
+                mutation.removedNodes.forEach((node) => {
+                    if ((node as HTMLElement).hasAttribute('data-vl-is-layer')) {
+                        this.vlMapLayers = this.vlMapLayers.filter(
+                            (layer) => (node as unknown as VlMapLayer).title !== layer.title
```

Comment:
als afnemer voor bepaalde reden geen `data-vl-name` instelt, wordt default `kaartlaag` genomen als title

als dit voor meerdere lagen gebeurt dan worden die allemaal tegelijk verwijderd, dus dit is niet de meest veilige manier.

Een manier waarop je het zou kunnen doen:

OpenLayers houd intern ook unique id's bij voor elke laag; stel om dan bv op volgende manier de juiste lagen er uit te halen (tegengekomen bij verdubbeling van map-layer-styles)

`(layer) => getUid(node as unknown as VlMapLayer) !== getUid(layer)`

dan moet je bij imports nog dit toevoegen: 
`import { getUid } from 'ol/util';`

Reply:
Very good spot thanks! :D

Ik heb het op een licht andere manier gefixt, ik vergelijk nu gewoon de layers als object met elkaar aangezien deze dezelfde referentie hebben.